internal/providers/pluginfw/common/common.go

```diff
@@ -45,3 +47,21 @@ func ConfigureResource(req resource.ConfigureRequest, resp *resource.ConfigureRe
 	}
 	return client
 }
+
+// GetDatabricksStagingName returns the resource name for a given resource with _pluginframework suffix.
+// Once a migrated resource is ready to be used as default, the Metadata method for that resource should be updated to use GetDatabricksProductionName.
+func GetDatabricksStagingName(name string) string {
+	return fmt.Sprintf("databricks_%s_pluginframework", name)
+}
+
+func GetDatabricksProductionName(name string) string {
+	return fmt.Sprintf("databricks_%s", name)
+}
+
+func SetResourceNameInContext(ctx context.Context, resourceName string) context.Context {
```

Comment:
To prevent `common.go` from being a dumping ground, let's put this into its own file, e.g. `useragent.go` or `context.go`.